pkg/names/generate.go

```diff
@@ -52,3 +56,10 @@ func (simpleNameGenerator) GenerateName(base string) string {
 	}
 	return fmt.Sprintf("%s-%s", base, utilrand.String(randomLength))
```

Comment:
i think there might actually be a bug here - the total length should be 63 or less, including the postfixed random string?

Reply:
Ah yeah good point, there is an off-by-one since the hyphen isn't accounted for in `maxGeneratedNameLength`, see https://github.com/knative/build-pipeline/pull/550/files#diff-0cd213308b34c799cb518fa9d6d1daacL46.

Reply:
Fixed!